tests/unit/unit_test_multipart_upload.py

```diff
@@ -114,3 +116,36 @@ def chunk_upload(part):
 
         # assert _put_chunk was called at least once
         assert_greater_equal(len(mocked_put_chunk.call_args_list), 1)
+
+
+def test__multipart_upload():
```

Comment:
test for multipart shouldn't care if it gets a single thread or multi-thread pool back. it should just use whatever the provider gives